test/unit/spec/signaling/v2/peerconnectionmanager.js

```diff
@@ -88,6 +160,56 @@ describe('PeerConnectionManager', () => {
     });
   });
 
+  describe('"connectionStateChanged"', () => {
+    it('emits only when the summarized connection state changes', async () => {
+      const test = makeTest();
+      await test.peerConnectionManager.createAndOffer();
+
+      // It should emit if a single PeerConnectionV2's .connectionState changes.
+      let didEmit = false;
+      test.peerConnectionV2s[0].connectionState = 'connecting';
+      test.peerConnectionManager.once('connectionStateChanged', () => { didEmit = true; });
+      test.peerConnectionV2s[0].emit('connectionStateChanged');
+      assert(didEmit);
+      assert.equal(test.peerConnectionManager.connectionState, 'connecting');
+
+      await test.peerConnectionManager.createAndOffer();
+
+      // It should not emit if a second PeerConnectionV2's .connectionState
+      // changes in a way that does not effect the summarized connection state.
+      didEmit = false;
+      test.peerConnectionManager.once('connectionStateChanged', () => { didEmit = true; });
+      test.peerConnectionV2s[1].connectionState = 'connecting';
+      test.peerConnectionV2s[1].emit('connectionStateChanged');
+      assert(!didEmit);
+      assert.equal(test.peerConnectionManager.connectionState, 'connecting');
+
+      // However, if a second PeerConnectionV2's .connectionState does cause
+      // a change in the summarized connection state, it _should_ emit
+      // "connectionStateChanged".
+      test.peerConnectionV2s[1].connectionState = 'connected';
+      test.peerConnectionV2s[1].emit('connectionStateChanged');
+      assert(didEmit);
+      assert.equal(test.peerConnectionManager.connectionState, 'connected');
+
+      // If a PeerConnectionV2's closing affects the summarized connection state,
+      // then it should emit.
+      didEmit = false;
+      test.peerConnectionManager.once('connectionStateChanged', () => { didEmit = true; });
+      test.peerConnectionV2s[1].state = 'closed';
+      test.peerConnectionV2s[1].emit('stateChanged', 'closed');
+      assert(didEmit);
+      assert.equal(test.peerConnectionManager.connectionState, 'connecting');
+
+      // Once the last PeerConnectionV2 is closed, and the summarized
+      // connection state was not "new", it should emit in the "new" state.
+      test.peerConnectionV2s[0].state = 'closed';
+      test.peerConnectionV2s[0].emit('stateChanged', 'closed');
+      assert(didEmit);
+      assert.equal(test.peerConnectionManager.connectionState, 'new');
```

Comment:
In C++, we emit closed when all PCs are closed. Ref: https://code.hq.twilio.com/client/video-sdk-api/blob/master/appendix/mediaconnectionstate.md

Reply:
How are we resolving the observed difference between C++ and JS?

Reply:
I re-read the IDL and it makes sense to me. If all the PCs are closed then the state should be closed.